src/include/hip/rpp_hip_common.hpp

```diff
@@ -9,207 +9,43 @@
-{
-    half2 x;
-    half2 y;
-} d_half4;
-typedef struct d_half6
-{
-    half2 x;
-    half2 y;
-    half2 z;
-} d_half6;
-typedef struct d_half8
-{
-    d_half4 x;
-    d_half4 y;
-} d_half8;
-typedef struct d_half24
-{
-    d_half8 x;
-    d_half8 y;
-    d_half8 z;
-} d_half24;
-typedef struct d_half4_as_halfs
-{
-    half x;
-    half y;
-    half z;
-    half w;
-} d_half4_as_halfs;
-typedef struct d_half8_as_halfs
-{
-    d_half4_as_halfs x;
-    d_half4_as_halfs y;
-} d_half8_as_halfs;
-typedef struct d_half24_as_halfs
-{
-    d_half8_as_halfs x;
-    d_half8_as_halfs y;
-    d_half8_as_halfs z;
-} d_half24_as_halfs;
-typedef struct d_half12_as_half3s
-{
-    d_half3 x;
-    d_half3 y;
-    d_half3 z;
-    d_half3 w;
-
-} d_half12_as_half3s;
-typedef struct d_half24_as_half3s
-{
-    d_half12_as_half3s x;
-    d_half12_as_half3s y;
-
-} d_half24_as_half3s;
+typedef struct { half h1[3];                                                                    }   d_half3_s;
+typedef struct { half2 h2[3];                                                                   }   d_half6_s;
+typedef union { half h1[8];     half2 h2[4];                                                    }   d_half8;
+typedef union { half h1[12];    half2 h2[6];    d_half3_s h3[4];                                }   d_half12;
+typedef union { half h1[24];    half2 h2[12];   d_half3_s h3[8];    d_half8 h8[3];              }   d_half24;
 
 // uchar
-typedef unsigned char uchar;
-typedef struct d_uchar8
-{
-    uchar4 x;
-    uchar4 y;
-} d_uchar8;
-typedef struct d_uchar24
-{
-    d_uchar8 x;
-    d_uchar8 y;
-    d_uchar8 z;
-} d_uchar24;
-typedef struct d_uchar12_as_uchar3s
-{
-    uchar3 x;
-    uchar3 y;
-    uchar3 z;
-    uchar3 w;
-
-} d_uchar12_as_uchar3s;
-typedef struct d_uchar24_as_uchar3s
-{
-    d_uchar12_as_uchar3s x;
-    d_uchar12_as_uchar3s y;
-
-} d_uchar24_as_uchar3s;
+typedef union { uchar uc1[8];   uchar4 uc4[2];                                                  }   d_uchar8;
+typedef union { uchar uc1[24];  uchar3 uc3[8];  d_uchar8 uc8[3];                                }   d_uchar24;
 
 // schar
-typedef signed char schar;
-typedef struct d_schar3
-{
-    schar x;
-    schar y;
-    schar z;
-} d_schar3;
-typedef struct d_schar4
-{
-    schar x;
-    schar y;
-    schar z;
-    schar w;
-} d_schar4;
-typedef struct d_schar8
-{
-    d_schar4 x;
-    d_schar4 y;
-} d_schar8;
-typedef struct d_schar24
-{
-    d_schar8 x;
-    d_schar8 y;
-    d_schar8 z;
-} d_schar24;
-typedef struct d_schar12_as_schar3s
-{
-    d_schar3 x;
-    d_schar3 y;
-    d_schar3 z;
-    d_schar3 w;
-
-} d_schar12_as_schar3s;
-typedef struct d_schar24_as_schar3s
-{
-    d_schar12_as_schar3s x;
-    d_schar12_as_schar3s y;
-
-} d_schar24_as_schar3s;
+typedef struct { schar sc1[8];                                                                  }   d_schar8_s;
+typedef struct { d_schar8_s sc8[3];                                                             }   d_schar24_s;
```

Comment:
This definition can cause confusion for the compiler. It might interpret it as 2dim array. Can you define it as schar xx[24] instead